analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -1022,6 +1024,96 @@ public void Main()
             validator.ValidateTag("MoreArgs", x => x.Should().BeNull());
         }
 
+        [DataTestMethod]
+        [DataRow("null", "false")]
+        [DataRow("new object()", "true")]
+        public void Invocation_DoesNotReturnIf_Continues(string value, string stopsIf)
+        {
+            var code = $@"
+public void Main()
+{{
+    Tag(""Begin"");
+    object value = {value};
+    CustomValidator(""Irrelevant"", value == null);
+    Tag(""End"");
+}}
+
+public void CustomValidator(object irrelevant, [System.Diagnostics.CodeAnalysis.DoesNotReturnIf({stopsIf})] bool condition) {{ }}";
+            var validator = SETestContext.CreateCSMethod(code).Validator;
+            validator.ValidateTagOrder("Begin", "End");
+            validator.ValidateExitReachCount(1);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DataRow("null", "true")]
+        [DataRow("new object()", "false")]
+        public void Invocation_DoesNotReturnIf_Stops(string value, string stopsIf)
+        {
+            var code = $@"
+public void Main()
+{{
+    Tag(""Begin"");
+    object value = {value};
+    CustomValidator(""Irrelevant"", value == null);
+    Tag(""Unreachable"");
+}}
+
+public void CustomValidator(object irrelevant, [System.Diagnostics.CodeAnalysis.DoesNotReturnIf({stopsIf})] bool condition) {{ }}";
+            var validator = SETestContext.CreateCSMethod(code).Validator;
+            validator.ValidateTagOrder("Begin");
+            validator.ValidateExitReachCount(0);
+            validator.ValidateExecutionCompleted();
+        }
+
+        [DataTestMethod]
+        [DataRow("true", "false")]
+        [DataRow("!false", "false")]
```

Comment:
I don't understand this. 
```
    var value = {value};
    CustomValidator(""Irrelevant"", value);
```
is same as
```
CustomValidator(""Irrelevant"", {value});
```
It just tests more likely scenario when the value comes from a symbol. Because `CustomValidator(true); // Always throws` is unlikely

Reply:
The difference is that `CustomValidator(""Irrelevant"", !false)` tests the `UnaryOperatorKind.Not` handling in ProcessAssertedBoolSymbol while `CustomValidator(""Irrelevant"", value);` does not. The assignment `value = !false;` tests some unrelated logic.

Reply:
The original uses unrelated scaffolding (assignment). The inlined tests something that is already tested from Debug.Assert. I've changed it, but I don't see to much difference in those.